workflow/controller/workflowpod.go

```diff
@@ -405,6 +406,85 @@ func (woc *wfOperationCtx) createWorkflowPod(ctx context.Context, nodeName strin
 		pod.Spec.Containers[i] = c
 	}
 
+	offloadEnvVarTemplate := false
+	for _, c := range pod.Spec.Containers {
+		if c.Name == common.MainContainerName {
+			for _, e := range c.Env {
+				if e.Name == common.EnvVarTemplate {
+					envVarTemplateValue = e.Value
+					if len(envVarTemplateValue) > maxEnvVarLen {
+						offloadEnvVarTemplate = true
+					}
+				}
+			}
+		}
+	}
+
+	if offloadEnvVarTemplate {
+		cmName := fmt.Sprintf("%s-configmap", pod.Name)
```

Comment:
can you logic to check name is not crossing the max limit ? if yes, we can short the name.